library/std/src/sys/unix/time.rs

```diff
@@ -293,7 +294,13 @@ mod inner {
 
     impl Instant {
         pub fn now() -> Instant {
-            Instant { t: now(libc::CLOCK_MONOTONIC) }
+            let t = match now(libc::CLOCK_BOOTTIME) {
+                Err(e) if e.kind() == io::ErrorKind::InvalidInput => now(libc::CLOCK_MONOTONIC),
+                v => v,
+            }
+            .unwrap();
```

Comment:
IMHO this match is easier to read than a `or_else`, but I'm not sure if that's faster/better/stronger.

Here's a godbolt link, maybe someone can decide what to use: https://rust.godbolt.org/z/6Eoxc7PEz

Reply:
There is not what **I** would consider a significant enough difference in the assembly to merit one approach over another, which is as it should be for such similar expressions.